plugins/modules/scc_product.py

```diff
@@ -72,6 +76,22 @@ def main():
     module.task_timeout = 4 * 60
 
     with module.api_connection():
+        scc_version = '1.0.0'  # fallback
+        try:
+            statuses = module.foremanapi.resource('ping').call('statuses')
+            plugins = statuses['results']['foreman']['plugins']
+            for plugin in plugins:
+                if 'foreman_scc_manager' in plugin:
+                    scc_version = plugin.split(',')[1]
+        except Exception:
+            pass
+
+        if LooseVersion(scc_version.strip()) < LooseVersion('1.8.6'):
+            scc_search_by = 'friendly_name'
+        else:
+            scc_search_by = 'name'
+
+        module.foreman_spec['scc_product']['search_by'] = scc_search_by
```

Comment:
This is a first, to my knowlege. We patch the foreman_spec on the fly that is completely in our controll.
:+1: anyway.

Reply:
It is, and it isn't, because when we write the spec, we never talked to the server ;)